### PR TITLE
[RW-4784][risk=no] Puppeteer project reorganizing dir structures

### PR DESCRIPTION
**Pull request by aweng98**
-  Files and directory reorganization.
-  Moved out common functions to its own file. `takeScreenshot`, `savePageToFile`, `makeString`, etc.

### CODE REVIEW

File: e2e/tests/app.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import {Page} from 'puppeteer';
-import GoogleLoginPage from 'app/google-login';
+import GoogleLoginPage from 'app/page/google-login';
 
 
 export const signIn = async (page: Page) => {
```
Comment (jmthibault79) [inline e2e/tests/app.ts:5]: should this also move to utils?

File: e2e/app/component/workspace-card.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import {ElementHandle, Page} from 'puppeteer';
 import {WorkspaceAccessLevel} from 'app/page-identifiers';
-import BaseElement from 'app/aou-elements/base-element';
```
Comment (jmthibault79) [inline e2e/app/component/workspace-card.ts:3]: WorkspaceCard extends BaseElement.  Is it a Component or an Element?

File: e2e/app/element/base-element.ts [hunk 3]
```diff
@@ -257,4 +257,29 @@ export default class BaseElement {
...
+    return await this.element.evaluate((element, k, v) => {
+      element.setAttribute(k, v);
+    }, key, value);
+  }
+
+  /**
+   * Paste texts instead type one char at a time.
+   * @param text
+   */
+  async paste(text: string) {
```
Comment (aweng98) [inline e2e/app/element/base-element.ts:275]: @calbach FYI, I got the textarea paste function to work finally.

File: e2e/tests/app.ts (hunk 1 above)
Comment (jmthibault79) [reply e2e/tests/app.ts:5]: sounds good to me!

File: e2e/app/component/workspace-card.ts (hunk 2 above)
Comment (jmthibault79) [reply e2e/app/component/workspace-card.ts:3]: ok!

File: e2e/app/page/workspace-edit-page.ts [hunk 4]
```diff
@@ -223,7 +223,7 @@ export const FIELD = {
       type: 'radiobutton',
       textOption: {textContains: LABEL_ALIAS.YES_FOCUS_ON_UNDERREPRESENTED_POPULATION}
     },
-    noOnUnderrepresentedPopulationRadiobutton: {
+    noUnderrepresentedPopulationRadiobutton: {
```
Comment (jmthibault79) [inline e2e/app/page/workspace-edit-page.ts:226]: what does this change do?

File: e2e/tests/home/home-page.spec.ts [hunk 5]
```diff
@@ -39,7 +39,7 @@ describe('Home page ui tests', () => {
       expect(cardName).toMatch(new RegExp(/^[a-zA-Z]+/));
 
       // check Workspace Action menu for listed actions
-      const ellipsis = await card.getEllipsis();
+      const ellipsis = card.getEllipsis();
```
Comment (jmthibault79) [inline e2e/tests/home/home-page.spec.ts:42]: is this correct?

File: e2e/tests/user/registration-ui.spec.ts [hunk 6]
```diff
@@ -72,6 +72,7 @@ describe('User registration tests:', () => {
     await (await createAccountPage.getTermsOfUseCheckbox()).check();
     const agreementPageButton = await createAccountPage.getNextButton();
     await agreementPageButton.clickWithEval();
+    await agreementPageButton.dispose();
```
Comment (jmthibault79) [inline e2e/tests/user/registration-ui.spec.ts:75]: please explain these 2 changes

File: e2e/tests/user/registration.spec.ts [hunk 7]
```diff
@@ -20,18 +20,21 @@ describe('User registration tests:', () => {
     let nextButton = await createAccountPage.getNextButton();
     await nextButton.waitUntilEnabled();
     await nextButton.click();
+    await nextButton.dispose();
```
Comment (jmthibault79) [inline e2e/tests/user/registration.spec.ts:23]: also these 3 changes

File: e2e/app/component/workspace-card.ts [hunk 8]
```diff
@@ -41,46 +45,58 @@ export default class WorkspaceCard extends BaseElement {
   }
 
   static async findCard(page: Page, workspaceName: string): Promise<WorkspaceCard | null> {
-    const selector = `.//*[@data-test-id="workspace-card-name" and text()="${workspaceName}"]`;
+    const selector = `.//*[@data-test-id="workspace-card-name" and normalize-space(text())="${workspaceName}"]`;
```
Comment (jmthibault79) [inline e2e/app/component/workspace-card.ts:48]: this can use the new SELECTOR instead

File: e2e/app/component/workspace-card.ts [hunk 9]
```diff
@@ -41,46 +45,58 @@ export default class WorkspaceCard extends BaseElement {
...
-      await card.dispose(); // not it, dispose the ElementHandle.
+      await handle.dispose(); // not it, dispose the ElementHandle.
     }
     return null; // not found
   }
 
 
-  constructor(page: Page) {
-    super(page);
+  constructor(private readonly page: Page) {
```
Comment (jmthibault79) [inline e2e/app/component/workspace-card.ts:62]: why this change?

File: e2e/app/component/workspace-card.ts [hunk 10]
```diff
@@ -41,46 +45,58 @@ export default class WorkspaceCard extends BaseElement {
...
+
+  }
+
+  async findCard(workspaceName: string): Promise<WorkspaceCard | null> {
+    const selector = `.//*[@data-test-id="workspace-card-name" and normalize-space(text())="${workspaceName}"]`;
```
Comment (jmthibault79) [inline e2e/app/component/workspace-card.ts:67]: use SELECTOR

File: e2e/app/component/workspace-card.ts (hunk 8 above)
Comment (aweng98) [reply e2e/app/component/workspace-card.ts]: 👍Done

File: e2e/app/component/workspace-card.ts (hunk 9 above)
Comment (aweng98) [reply e2e/app/component/workspace-card.ts:62]: Because class is no longer extends BaseElement.

File: e2e/app/component/workspace-card.ts (hunk 10 above)
Comment (aweng98) [reply e2e/app/component/workspace-card.ts]: Done

File: e2e/app/page/workspace-edit-page.ts (hunk 4 above)
Comment (aweng98) [reply e2e/app/page/workspace-edit-page.ts:226]: typo. ui text string does not contain `on` word.

File: e2e/tests/home/home-page.spec.ts (hunk 5 above)
Comment (aweng98) [reply e2e/tests/home/home-page.spec.ts:42]: yes, correct.  It's non-asynchronous method.

File: e2e/tests/user/registration-ui.spec.ts (hunk 6 above)
Comment (aweng98) [reply e2e/tests/user/registration-ui.spec.ts]: `dispose()`: change the object to unusable explicitly instead implicitly relying on Puppeteer tool to do the same job.

File: e2e/app/page/workspace-edit-page.ts (hunk 4 above)
Comment (jmthibault79) [reply e2e/app/page/workspace-edit-page.ts:226]: ok

File: e2e/tests/home/home-page.spec.ts (hunk 5 above)
Comment (jmthibault79) [reply e2e/tests/home/home-page.spec.ts:42]: ok

File: e2e/tests/user/registration-ui.spec.ts (hunk 6 above)
Comment (jmthibault79) [reply e2e/tests/user/registration-ui.spec.ts]: ok
